zeroex/orderwatch/decoder/event_decoder.go

```diff
@@ -574,7 +584,13 @@ func (d *Decoder) decodeERC721(log types.Log, decodedLog interface{}) error {
 
 	err := unpackLog(decodedLog, eventName, log, d.erc721ABI)
 	if err != nil {
-		return err
+		if _, ok := err.(UnsupportedEventError); ok {
+			// Try unpacking using the incorrect ERC721 event ABIs
+			err := unpackLog(decodedLog, eventName, log, d.incorrectERC721EventsABI)
+			if err != nil {
+				return err
```

Comment:
It might be helpful to combine the two error messages here in order to get a clearer picture of what went wrong.

Something like:

```go
	erc721Err := unpackLog(decodedLog, eventName, log, d.erc721ABI)
	if erc721Err != nil {
		if _, ok := erc721Err.(UnsupportedEventError); ok {
			// Try unpacking using the incorrect ERC721 event ABIs
			fallbackErr := unpackLog(decodedLog, eventName, log, d.incorrectERC721EventsABI)
			if fallbackErr != nil {
				return fmt.Errorf("could not decode event with either correct ERC721 ABI or fallback ABI: %s %s", erc721Err.Error(), fallbackErr.Error())
			}
		}
	}
```

I'm not sure the best way to name these error variables or phrase the error message.

Reply:
We need to throw the `UnsupportedEventError` type if it is an unsupported event since we want to log these to then add custom support for other incorrect standard implementations. Because of this, I don't think this makes a lot of sense. All the caller needs to know if that the we were unable to decode the event given the logic in place.

Reply:
@fabioberger the problem with the way it is currently written is that we *don't* return the error that is associated with trying to unpack the event as a standard ERC721 event. We _only_ are returning the error associated with trying to unpack the event using `incorrectERC721EventsABI`. I'm worried that this could obscure some issues.